src/test/java/com/github/lombrozo/jsmith/antlr/rules/RulerefTest.java

```diff
@@ -54,4 +55,46 @@ void generatesRuleReferenceByUsingLink() throws WrongPathException {
             Matchers.equalTo(expected)
         );
     }
+
+    @Test
+    void generatesRuleReferenceByUsingArgActionBlock() throws WrongPathException {
+        final String ref = "reference";
+        final ArgActionBlock actionblock = new ArgActionBlock(new Root(), new ArrayList<>(0));
+        final String expected = "Linked rule value(1)";
+        final String expectedtwo = "Linked rule value(2)";
+        actionblock.append(new Literal(expected));
+        actionblock.append(new Literal(expectedtwo));
+        final Unparser unparser = new Unparser();
+        unparser.with(ref, actionblock);
+        final Ruleref ruleref = new Ruleref(new Empty(), ref, unparser);
+        MatcherAssert.assertThat(
+            String.format(
+                "We expect that %s will invoke the linked argActionBlock to generate output",
+                ruleref
+            ),
+            ruleref.generate(new Context()).text().output(),
+            Matchers.equalTo(expected.concat(expectedtwo))
+        );
+    }
+
+    @Test
+    void generatesRuleReferenceByUsingElementOptions() throws WrongPathException {
+        final String ref = "reference";
+        final ElementOptions elemopts = new ElementOptions(new Empty());
+        final String expected = "Linked rule value";
+        final ElementOption elemopt = new ElementOption(elemopts);
+        elemopt.append(new Identifier(elemopt, expected));
+        elemopts.append(elemopt);
```

Comment:
@RechnoyBobr Could you use a single noun for variable names, please? https://www.yegor256.com/2015/01/12/compound-name-is-code-smell.html